src/odemis/acq/stream_settings.py

```diff
@@ -0,0 +1,174 @@
+            try:
+                with open(self.file_path, 'r') as jsonfile:
+                    self.config_data = json.load(jsonfile)
+            except json.decoder.JSONDecodeError as e:
+                logging.warning(f"Error decoding JSON: {e}")
+            except Exception as e:
+                logging.warning(f"Error in reading the stream settings: {e}")
+
+    def _write(self):
+        logging.debug("Writing the most recently used stream settings in JSON")
+        try:
+            if self.config_data:
+                with open(self.file_path, 'w') as jsonfile:
+                    json.dump(self.config_data, jsonfile, indent=2)
+        except Exception as e:
+            logging.warning(f"Error in writing the stream settings: {e}")
+
+    def update_data(self, new_entries: List[Dict[str, Any]]):
+        """
+        Update the saved stream setting with the list of new_entries. First element in JSON file
+        is the latest entry while the last element is the oldest stream setting. The file updates
+        its entries based on the new_entries while maintaining a fixed length of entries in the file.
+        :param new_entries: (list of stream controller settings) Each element in the list represent the setting
+         of the stream saved as a list from oldest to latest stream setting
+        """
+        # Convert config data to list to enable list operations
+        if self.config_data is None:
+            self.config_data = []
+        data = self.config_data
+
+        for e in new_entries:
+            # find if the new entry name exists in the saved data
+            index = get_config_index(data, e["name"])
+            # pop the entry from the list if the stream name is same to avoid duplication
+            if index is not None:
+                data.pop(index)
+            # insert the new entry at the first position
+            data.insert(0, e)
+            # length of list of entries should be constant
+            data = data[:self.max_entries]
+
+        self.config_data = data
+        self._write()
+
+    def get_settings_order(self, stream: FluoStream):
+        """
+        Get the VAs for loading a stream setting and sort them so it can applied in the given order while loading
+        a stream later.
+        :param stream: stream object that consists of VAs
+        """
+        if self.settings_order is None:
+            stream_vas = model.getVAs(stream)
+            settings = set(stream_vas.keys()).difference(NON_SETTINGS_VA)
+            self.settings_order = util.sorted_according_to(settings, SETTINGS_ORDER)
+
+    def get_streams_settings(self, streams: list):
+        """
+        Get the streams settings from the stream controller to update the entries in the JSON file.
+        :param streams: (list of streams) List of vigilant attributes of each stream settings
+        """
+        local_entries = []
+
+        if len(streams) > 0:
+            self.get_settings_order(streams[0])
```

Comment:
You should get the order per stream. Right now you always call this function with the same type of stream, so that works, but it's best to no rely on it. => then you can remove the `if len(...) > 0`.